### PR TITLE
feat: combine user registration endpoints

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #https://github.com/lightdash/lightdash/issues/6866 

### Description:

Main changes:

- moved **api/v1/user [Get]** to UserController
- moved and combined **api/v1/user [Post]** and **api/v1/register [Post]** to UserController
  - in the app this would correspond to the register action from `/register` and to accept an invite `/invite`

Smaller changes:

- Added email type so the API can validate it
- Amended some typing around user creation / activation ( haven't renamed all types as that would increase a lot the PR scope )
- Update e2e tests and FE with new types and endpoint

### CODE REVIEW

**Review by joaoviana — APPROVED** [review]
Looks great!

File: packages/common/src/types/api/email.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/**
+ * Email
+ * @pattern ^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$
+ */
+export type Email = string;
```
Comment (joaoviana) [inline packages/common/src/types/api/email.ts:5]: only thing I don't understand is how this pattern is consumed 🤔

Comment (ZeRego) [reply packages/common/src/types/api/email.ts:5]: Is consumed by `tsoa` lib when we run `yarn workspace backend generate-api` which generates files that validate API requests and documentation. 

Docs on tsoa @pattern annotation https://tsoa-community.github.io/docs/annotations.html

Comment (joaoviana) [reply packages/common/src/types/api/email.ts:5]: tyyy!!! makes sense

**Pull request merged**